packages/core/useOTP/index.ts

```diff
@@ -0,0 +1,74 @@
+// eslint-disable-next-line no-restricted-imports
+import type { Ref } from 'vue'
+import { onBeforeUnmount, ref } from 'vue-demi'
+
+/**
+ * Type definition for the composable return values
+ */
+interface UseOTPAutoFillReturn {
+  otp: Ref<string>
+  fetchOTP: () => void
+}
+
+/**
+ * Composable to handle OTP autofill using the OTPCredentials API
+ * This composable automatically requests the OTP via SMS when mounted
+ * and provides a ref that contains the OTP value.
+ */
+export function useOTP(): UseOTPAutoFillReturn {
+  const otp = ref<string>('') // Ref to store the OTP value
+  const abortController = ref<AbortController | null>(null) // To abort OTP requests if needed
+
+  /**
+   * Internal function to handle the filled OTP and set it to the ref
+   * @param otpCode The OTP code received
+   */
+  const handleOTPFilled = (otpCode: string) => {
+    otp.value = otpCode
+  }
+
+  /**
+   * Function to request OTP via the OTPCredentials API
+   */
+  const fetchOTP = () => {
+    // Check if the browser supports the OTPCredentials API
+    if (!('OTPCredential' in window)) {
+      console.warn('OTPCredentials API is not supported by this browser.')
```

Comment:
HOw about also returning a boolean by the composable to specify this is supported. So users can also do something else if the browserr isn't supported ?

Reply:
@huang-julien Great idea! I think this will be very useful.